SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp

```diff
@@ -238,18 +243,32 @@ void USpatialReceiver::OnRemoveEntity(const Worker_RemoveEntityOp& Op)
 		return;
 	}
 
-	RemoveActor(Op.entity_id);
+	if (LoadBalanceEnforcer != nullptr)
+	{
+		LoadBalanceEnforcer->OnEntityRemoved(Op);
+	}
+
 	OnEntityRemovedDelegate.Broadcast(Op.entity_id);
 }
 
 void USpatialReceiver::OnRemoveComponent(const Worker_RemoveComponentOp& Op)
 {
+	if (Op.component_id == SpatialConstants::UNREAL_METADATA_COMPONENT_ID)
+	{
+		RemoveActor(Op.entity_id);
```

Comment:
If this entity is waiting on async load, we should remove it from `EntitiesWaitingForAsyncLoad` and not call `RemoveActor`.

Reply:
done